legal-api/src/legal_api/services/filings/validations/incorporation_application.py

```diff
@@ -226,6 +234,59 @@ def validate_incorporation_effective_date(incorporation_json) -> Error:
 
     return None
 
+def validate_cooperative_documents(incorporation_json) -> Error:
+    """Return an error or warning message based on the cooperative documents validation rules.
+
+    Rules:
+        - The documents are provided.
+        - Document IDs are unique.
+    """
+    # Setup
+    msg = []
+
+    rules_file_key = incorporation_json['filing']['incorporationApplication']['cooperative']['rulesFileKey']
+    rules_file_name = incorporation_json['filing']['incorporationApplication']['cooperative']['rulesFileName']
+    memorandum_file_key = incorporation_json['filing']['incorporationApplication']['cooperative']['memorandumFileKey']
+    memorandum_file_name = incorporation_json['filing']['incorporationApplication']['cooperative']['memorandumFileName']
+
+    # Validate key values exist
+    if not rules_file_key:
+        msg.append({'error': babel('A valid rules key is required.')})
+
+    if not rules_file_name:
+        msg.append({'error': babel('A valid rules file name is required.')})
+
+    if not memorandum_file_key:
+        msg.append({'error': babel('A valid memorandum key is required.')})
+
+    if not memorandum_file_name:
+        msg.append({'error': babel('A valid memorandum file name is required.')})
+
+    if msg:
+        return msg
+
+    # Validate the files from Minio
+    try:
+        MinioService.get_file(rules_file_key)
+        file_info = MinioService.get_file_stat_object(rules_file_key)
+        if file_info.size > 10000:
+            msg.append({'error': babel('Rules file exceeds maximum size.')})
+    except Exception:
+        msg.append({'error': babel('Invalid rules file.')})
+
+    try:
+        memorandum_file = MinioService.get_file(memorandum_file_key)
```

Comment:
Do we need this get?

Reply:
We will once i figure out how to read/verify this 👍